src/web/tokenserver.rs

```diff
@@ -1,26 +1,165 @@
+    // the public rsa components come from
+    // https://oauth.accounts.firefox.com/v1/jwks
+    // TODO we should fetch it from an environment var instead of hardcoding it.
+    let token_data = decode::<Claims>(
+        &auth.token(),
+        &DecodingKey::from_rsa_components("2lDphW0lNZ4w1m9CfmIhC1AxYG9iwihxBdQZo7_6e0TBAi8_TNaoHHI90G9n5d8BQQnNcF4j2vOs006zlXcqGrP27b49KkN3FmbcOMovvfesMseghaqXqqFLALL9us3Wstt_fV_qV7ceRcJq5Hd_Mq85qUgYSfb9qp0vyePb26KEGy4cwO7c9nCna1a_i5rzUEJu6bAtcLS5obSvmsOOpTLHXojKKOnC4LRC3osdR6AU6v3UObKgJlkk_-8LmPhQZqOXiI_TdBpNiw6G_-eishg8V_poPlAnLNd8mfZBam-_7CdUS4-YoOvJZfYjIoboOuVmUrBjogFyDo72EPTReQ", "AQAB"),
+        &Validation::new(Algorithm::RS256),
+    ).map_err(|ee| {
+        ApiError::from(ApiErrorKind::Internal(format!("Unable to decode token_data: {:}", ee)))
+    })?;
+    println!("token data! {:?}", token_data);
+    let email = format!("{:}@api.accounts.firefox.com", token_data.claims.sub);
+
+    // TODO pull out of settings instead
+    let shared_secret = env::var("SYNC_MASTER_SECRET").expect("SYNC_MASTER_SECRET must be set");
+    let database_url =
+        env::var("TOKENSERVER_DATABASE_URL").expect("TOKENSERVER_DATABASE_URL must be set");
+
+    let connection = MysqlConnection::establish(&database_url)
+        .unwrap_or_else(|_| panic!("Error connecting to {}", database_url));
+    let user_record = diesel::sql_query(
+        "select users.uid, services.pattern, users.email, users.generation, users.client_state, users.created_at, users.replaced_at, nodes.node, users.keys_changed_at from users, services, nodes where users.email = ? and services.id = users.service and nodes.id = users.nodeid and nodes.service = services.id")
+        .bind::<Text, _>(email)
+        .load::<TokenserverUser>(&connection).unwrap();
+    println!("user record!! {:?}", user_record);
+    let (python_result, python_derived_result) = Python::with_gil(|py| {
+        let tokenlib = PyModule::from_code(
+            py,
+            r#"
+import tokenlib
+def make_token(plaintext, shared_secret):
+    return tokenlib.make_token(plaintext, secret=shared_secret)
+
+def get_derived_secret(plaintext, shared_secret):
+    return tokenlib.get_derived_secret(plaintext, secret=shared_secret)
+
+"#,
+            "main.py",
+            "main",
+        )
+        .map_err(|e| {
+            e.print_and_set_sys_last_vars(py);
+            e
+        })?;
+        let thedict = [
+            ("node", user_record[0].node.as_ref()),
+            ("uid", token_data.claims.sub.as_ref()),
+            ("fxa_kid", "asdf"), // userid component of authorization email
+            ("fxa_uid", "qwer"),
+            ("hashed_device_id", "..."),
+            ("hashed_fxa_uid", "..."),
+        ]
+        .into_py_dict(py);
+        // todo don't hardcode
+        // we're supposed to check the "duration" query
+        // param and use that if present (for testing)
+        thedict.set_item("expires", 300).unwrap();
```

Comment:
I'd at least make `300` a const, since it's used later as well.

Reply:
I'm going to leave this for now and do it in bug #898